### PR TITLE
feat: add gltf decoder

### PR DESCRIPTION
**Pull request by MPanknin**
Custom decoder for supporting the glTF data layout

### CODE REVIEW

File: example/tsconfig.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "alwaysStrict": true,
-    "target": "es6",
+    "target": "es2020",
```
Comment (MPanknin) [inline example/tsconfig.json:15]: Needed for `BigInt` numeric values

File: src/loading2/worker-pool.ts [hunk 2]
```diff
@@ -14,14 +16,17 @@ function createWorker(type: WorkerType): Worker {
 	case WorkerType.DECODER_WORKER: {
 		return new DecoderWorker();
 	}
+	case WorkerType.DECODER_WORKER_GLTF: {
+		return new DecoderWorker_GLTF();
+	}
```
Comment (pjurczynski) [inline src/loading2/worker-pool.ts:21]:
```suggestion
	}
```

File: src/loading2/gltf-decoder.worker.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+			};
+
+		}
+
+	}
+
+	// let duration = performance.now() - tStart;
+	// let pointsPerMs = numPoints / duration;
+	// console.log(`duration: ${duration.toFixed(1)}ms, #points: ${numPoints}, points/ms: ${pointsPerMs.toFixed(1)}`);
+
```
Comment (pjurczynski) [inline src/loading2/gltf-decoder.worker.js:127]:
```suggestion
```

File: src/loading2/load-octree.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ export async function loadOctree(
 	xhrRequest: XhrRequest,
 ) {
 	const trueUrl = await getUrl(url);
-
+
```
Comment (pjurczynski) [inline src/loading2/load-octree.ts:10]:
```suggestion

```

File: src/loading2/worker-pool.ts [hunk 5]
```diff
@@ -14,14 +16,17 @@ function createWorker(type: WorkerType): Worker {
 	case WorkerType.DECODER_WORKER: {
 		return new DecoderWorker();
 	}
+	case WorkerType.DECODER_WORKER_GLTF: {
+		return new DecoderWorker_GLTF();
```
Comment (pjurczynski) [inline src/loading2/worker-pool.ts:20]: Would it be possible to move the `require` here? That way if client does use only one type of the decoder, they won't over-import.

**Review by pjurczynski — APPROVED** [review]

**Pull request merged**